Jenkinsfile

```diff
@@ -1,43 +1,61 @@
-node('docker') {
-
-    docker.image('cambuilder:latest').inside('-u root') {
-        stage 'Cleanup workspace'
-        sh 'chmod 777 -R .'
-        sh 'rm -rf *'
+pipeline {
+    agent {
+        label 'cambuilder'
+    }
 
-        stage 'Checkout SCM'
-            checkout([
-                $class: 'GitSCM',
-                branches: [[name: "refs/heads/${env.BRANCH_NAME}"]],
-                extensions: [[$class: 'LocalBranch']],
-                userRemoteConfigs: scm.userRemoteConfigs,
-                doGenerateSubmoduleConfigurations: false,
-                submoduleCfg: []
-            ])
+    stages {
+        stage('Checkout SCM') {
+            steps{
+                checkout([
+                    $class: 'GitSCM',
+                    branches: [[name: "refs/heads/${env.BRANCH_NAME}"]],
+                    extensions: [[$class: 'LocalBranch']],
+                    userRemoteConfigs: scm.userRemoteConfigs,
+                    doGenerateSubmoduleConfigurations: false,
+                    submoduleCfg: []
+                ])
+            }
+        }
 
-        stage 'Install & Unit Tests'
-            timestamps {
-                timeout(time: 30, unit: 'MINUTES') {
-                    try {
+        stage('Install & Unit Tests') {
+            options {
+                timestamps()
+                timeout(time: 30, unit: 'MINUTES') 
+            }
+            steps
+                {
                         sh 'pip install . -U --pre'
```

Comment:
LGTM

As a side note, perhaps we should move away from running tests with `python setup.py nosetests` instead use `tox`.
Look at https://github.com/ska-sa/katconf/blob/master/Jenkinsfile

But for now, looks good.

Reply:
@mmphego We will move to tox as we do the Python 2 to 3 migration.